src/ja/common/proxy/remote.py

```diff
@@ -6,9 +6,14 @@ class Remote(ABC):
     Abstract base class for remote objects accessed by Proxy objects.
     Remotes are single-class command line programs.
 
-    Remotes receive a Message object represented by a YAML string on stdin.
-    Remotes then convert the YAML string to a Message object. Remotes then pass
-    on the Message object to a MessageHandler object via a socket. Remotes then
-    listen for a response (YAML string) on a socket. The response is then
-    written to stdout.
+    Receives a Command object represented by a YAML string on stdin. Asserts
+    that the user specified in the YAML document is the user calling the
+    Remote. Passes on the YAML string to a CommandHandler object via a socket.
+    Listens for a Response (as YAML string) on a different socket. Writes the
+    Response YAML string to stdout.
     """
+    def __init__(self, output_port: int, input_port: int):
```

Comment:
Again, ports or sockets? Also, to connect to a string, you need a string (when you create a socket, you get an FD, which you then bind to a file on the filesystem. When connecting to this socket, you need to provide the name, because you can't have an FD)